torchaudio/functional.py

```diff
@@ -1193,6 +1193,52 @@ def contrast(
     return output_waveform
 
 
+def dcshift(
+        waveform: Tensor,
+        shift: float,
+        limiter_gain: float = 0.
+) -> Tensor:
+    r"""Apply a DC shift to the audio. Similar to SoX implementation.
+    This can be useful to remove a DC offset
+    (caused perhaps by a hardware problem in the recording chain) from the audio
+
+    Args:
+        waveform (Tensor): audio waveform of dimension of `(..., time)`
+        shift (float): indicates the amount to shift the audio
+            Allowed range of values for shift : -2.0 to +2.0
+        limiter_gain (float): It is used only on peaks to prevent clipping
+            It should have a value much less than 1 (e.g. 0.05 or 0.02)
+
+    Returns:
+        Tensor: Waveform of dimension of `(..., time)`
+
+    References:
+        http://sox.sourceforge.net/sox.html
+    """
+    output_waveform = waveform
+    limiter_threshold = 0.
+    use_limiter = False
+
+    if limiter_gain != 0.0:
+        use_limiter = True
+        limiter_threshold = 1.0 - (abs(shift) - limiter_gain)
+
+    if use_limiter and shift > 0:
+        mask = waveform > limiter_threshold
+        temp = (waveform[mask] - limiter_threshold) * limiter_gain / (1 - limiter_threshold)
+        output_waveform[mask] = (temp + limiter_threshold + shift).clamp(max=limiter_threshold)
+        output_waveform[~mask] = (waveform[~mask] + shift).clamp(min=-1, max=+1)
```

Comment:
I cannot explain why but this seems to resolve torch script test issue.

```suggestion
        output_waveform[~mask] = (waveform[~mask] + shift).clamp(min=-1, max=1)
```